lib/galaxy/files/sources/__init__.py

```diff
@@ -267,6 +267,8 @@ class FilesSource(SingleFileSource, SupportsBrowsing):
     implements the `SupportsBrowsing` interface.
     """
 
+    plugin_type: ClassVar[str]
```

Comment:
This doesn't seem needed, as it is already present in the only subclass `BaseFilesSource` below (and I couldn't spot relevant changes in the rest of the `file_source_templates` branch).

Reply:
Well I didn't want to use the base class (BaseFileSources) for the other typing - I think things should program against the interface and objects should be instantiated against the concrete class (https://stackoverflow.com/questions/2697783/what-does-program-to-interfaces-not-implementations-mean). I started to go down that route and noticed the rest of the plumbing was already using the concrete class. This was a first step toward fixing that.

Reply:
I don't mind either way, but then maybe remove the same line from `BaseFilesSource` ?